### ISSUE TITLE
[JavaScript] Suggestion inconsistencies with parameter documentation

### ISSUE THREAD

**Issue by parasyte** [inch-ci/inch_ci-web#158]
Suggestions on [me.Tween#start](http://inch-ci.org/github/melonjs/melonjs/revision/ea876d25/code_object/2769555?branch=master) want a private parameter described.

A private parameter on a different function is not given the same suggestion treatment: [me.Sprite.setCurrentAnimation](http://inch-ci.org/github/melonjs/melonjs/revision/ea876d25/code_object/2759972?branch=master) See the [source code](https://github.com/melonjs/melonJS/blob/50597e86b39e5dd7a0efaf8aa74de1142013ffaa/src/renderable/sprite.js#L334-L373); `_preserve_dt` is not documented, and Inch-CI does not make a suggestion for it.

---
Re: [B] [me.Tween#start](http://inch-ci.org/github/melonjs/melonjs/revision/ea876d25/code_object/2769555?branch=master) in [melonjs/melonJS](http://inch-ci.org/github/melonjs/melonjs)

**Comment by rrrene** [comment]
I see why this is happening. Just out of curiousity: what are private parameters to a function?

**Comment by parasyte** [comment]
@rrrene TBH, I haven't seen anything in the JSDoc spec about it. But as a convention, we use a leading underscore on parameter names that are optional and undocumented for various reasons, like overriding default functionality for internal use.

In the case of the tween.start library call, the private parameter allows the tween engine to start a tween with an absolute timestamp; it defaults to using the current time, and there's really no reason for public callers to use it.

Anyway, that's the explanation for what I mean by "private parameter." Whichever way you choose to fix it it fine, though; either add the suggestion to all cases where params are not documented, or make an exception for param names with leading underscore.

**Comment by rrrene** [comment]
Thanks for the explanation. :+1: But I am not sure why there is an issue here: The current behaviour is what you describe in your last sentence: Parameters with a leading underscore are considered "not for public use" and won't trigger a documentation suggestion.

Am I misunderstanding something?

**Comment by parasyte** [comment]
Yep, have a look at [me.Tween#start](http://inch-ci.org/github/melonjs/melonjs/revision/ea876d25/code_object/2769555?branch=master) Including a screenshot below, with emphasis added:

<img width="512" alt="screen shot 2016-12-14 at 10 36 14 pm" src="https://cloud.githubusercontent.com/assets/456942/21214185/e63fa3a2-c24d-11e6-91b7-dfad0362cb80.png">

**Comment by rrrene** [comment]
Ouch. My mistake. Sorry, lack of coffee! 😢

**Comment by parasyte** [comment]
:+1: No worries. I'm kind of wired at the moment. :)
